hr_timesheet_hours_billed/models/hours_billed.py

```diff
@@ -0,0 +1,45 @@
+from odoo import api, fields, models
+
+
+class AccountAnalyticLine(models.Model):
+    _inherit = "account.analytic.line"
+
+    approved = fields.Selection(
+        selection=[("no", "No"), ("yes", "Yes")],
+        default="no",
+        required=True,
+    )
+    approved_uid = fields.Many2one(
+        "res.users",
+        string="Approved by",
+        index=True,
+        tracking=True,
+        compute="_compute_approved_uid",
```

Comment:
You can use one compute function for both methods because they are always triggered at the same time